### PR TITLE
Add faster entry points for drawPoints and friends

### PR DESCRIPTION
**Pull request by abarth**
Fixes https://github.com/flutter/flutter/issues/10087
Fixes https://github.com/flutter/flutter/issues/10233

### CODE REVIEW

**Comment by jason-simmons** [comment]
LGTM

File: lib/ui/painting.dart [hunk 1]
```diff
@@ -1174,13 +1174,31 @@ class Vertices extends NativeFieldWrapperClass2 {
...
       _encodePointList(textureCoordinates) : null;
     Int32List encodedColors = (colors != null) ? _encodeColorList(colors) : null;
     Int32List encodedIndices = (indices != null) ? new Int32List.fromList(indices) : null;
 
     _constructor();
-    _init(mode.index, encodedPositions, encodedTexs, encodedColors, encodedIndices);
+    _init(mode.index, encodedPositions, encodedTextureCoordinates, encodedColors, encodedIndices);
+  }
+
+  Vertices.raw(
```
Comment (Hixie) [inline lib/ui/painting.dart]: docs?

File: lib/ui/painting.dart [hunk 2]
```diff
@@ -1174,13 +1174,31 @@ class Vertices extends NativeFieldWrapperClass2 {
...
+    VertexMode mode,
+    Float32List positions, {
+    Float32List textureCoordinates,
+    Int32List colors,
+    Int32List indices,
+  }) {
+    if (textureCoordinates != null && textureCoordinates.length != positions.length)
+      throw new ArgumentError("[positions] and [textureCoordinates] lengths must match");
```
Comment (Hixie) [inline lib/ui/painting.dart]: Can you remove the square brackets, maybe replace them with double quotes? I think they come from an earlier time where we weren't sure about the style.

Also, periods at the end of the exception strings? ArgumentError doesn't say what the convention is but FlutterError says to be fully punctuated: https://docs.flutter.io/flutter/foundation/FlutterError/message.html

File: lib/ui/painting.dart [hunk 3]
```diff
@@ -1615,11 +1653,17 @@ class Canvas extends NativeFieldWrapperClass2 {
   void drawVertices(Vertices vertices, BlendMode blendMode, Paint paint) {
     _drawVertices(vertices, blendMode, paint._objects, paint._data);
   }
+
   void _drawVertices(Vertices vertices,
                      BlendMode blendMode,
                      List<dynamic> paintObjects,
                      ByteData paintData) native "Canvas_drawVertices";
 
+  ///
```
Comment (Hixie) [inline lib/ui/painting.dart]: please comment these out so that they come up on our list of missing docs

File: lib/ui/painting.dart [hunk 4]
```diff
@@ -1663,6 +1707,44 @@ class Canvas extends NativeFieldWrapperClass2 {
       colorBuffer, blendMode.index, cullRectBuffer
     );
   }
+
+  ///
```
Comment (Hixie) [inline lib/ui/painting.dart]: ditto

File: lib/ui/painting.dart (hunk 2 above)
Comment (Hixie) [inline lib/ui/painting.dart]: there's several examples of these, please audit all calls to `new ArgumentError`

**Comment by Hixie** [comment]
is there any way to test this?

**Comment by abarth** [comment]
Done.  PTAL

**Comment by Hixie** [comment]
LGTM

**Pull request merged**